modules/cache.py

```diff
@@ -37,7 +38,12 @@ def add(self, url:str):
             logging.info(f"Video size ({video.filesize} bytes) exceeds max cache size ({self.max_size_bytes} bytes). Caching cancelled.")
             return None 
         video_file_name = video.default_filename
+        start_time = time.perf_counter()
         video.download(self.file_path)
+        download_time = time.perf_counter() - start_time
+        MetricsHandler.download_time.inc(download_time)
```

Comment:
make download time a summary, then you can use the library's built in `time` function to time the code for you
```py
with MetricsHandler.download_time.time():
  video.download(self.file_path)
```
https://prometheus.github.io/client_python/instrumenting/summary/